### ISSUE TITLE
TMPDIR could be /data/data/com.termux/cache

### ISSUE THREAD

**Issue by shawnl** [termux/termux-packages#1772]
as documented here: https://github.com/aosp-mirror/platform_bionic/blob/master/libc/bionic/tmpfile.cpp#L87

It would also be nice if lineageos actually implemented tmpfile()..and then it got merged for future android versions..but I'm not holding out on that one. Android is not a normal open source project.

**Comment by xeffyr** [comment]
Yes, it can be. But what benefits we get when TMPDIR will be changed ?

**Comment by vishalbiswas** [comment]
@xeffyr we can just clear cache and it will clear all tmp files too

**Comment by xeffyr** [comment]
I think that this should be ok. But '/tmp' directory should be present for compatibility reasons or it should be a symlink to cache directory.

**Comment by shawnl** [comment]
"Clear cache" in the android app settings will work

**Comment by xeffyr** [comment]
@shawnl I know, I've just tested this.

**Comment by shawnl** [comment]
We can't provide /tmp cause that would require root.

**Comment by xeffyr** [comment]
'/tmp' in termux is '$PREFIX/tmp'.

**Comment by vishalbiswas** [comment]
and we should probably introduce a `TERMUX_TMPDIR` variable in build-package.sh

**Comment by fornwall** [comment]
This sounds nice.

We could perhaps make `$PREFIX/tmp` a symlink to `/data/data/com.termux/cache`, so that the general `/folder -> $PREFIX/folder` mapping still holds?
